### ISSUE TITLE
Problems with search with filter on 3rd position

### ISSUE THREAD

**Issue by Shaihuludus** [Adobe-Marketing-Cloud/asset-share-commons#239]
3rd search filter of any type is not working

**Environment**
* AEM Version: 6.4.1
* Asset Share Commons Version: 1.6.3-SNAPSHOT and 1.6.2
* Author, Publish or both? both

**To Reproduce**
1. Go to 'Page with Search Results' - it can be demo page from Asset Share Commons
2. Use 3 search filter (in this case created date but any filter treated as 3rd one and set as 3_group will cause the same problem)
4. Filter is not working

**Expected behavior**
Asset should be filtered

**Additional context**
There is some strange behaviour in adding predicates to query in backend. It adds paths from search result configuration as 3_group and with p.or set to true. So if we will have any query parameter from search coming also as 3_group

Fragment from debbuger for better problem description:

"3_group.0_path" -> "/content/dam/someassets"
"3_group.created.lowerBound" -> "2018-8-9"
"3_group.daterange.property" -> "jcr:created"
"3_group.daterange.upperBound" -> "2018-8-18"
"3_group.p.or" -> "true"

**Comment by davidjgonzalez** [comment]
@Shaihuludus by any chance have you tried this on 6.4.0? I just tried a build from develop, and it seems to work for me:


```
25.08.2018 19:52:39.211 *DEBUG* [0:0:0:0:0:0:0:1 [1535241159195] GET /content/asset-share-commons/en/light.results.html HTTP/1.1] com.day.cq.search.impl.builder.QueryImpl executing query (URL):
10_group.orderby.sort=desc&1_group.propertyvalues.0_values=image%2fjpeg%2cimage%2fgif%2cimage%2fpng%2cimage%2ftiff%2cimage%2fvnd.adobe.photoshop%2capplication%2fpostscript&1_group.propertyvalues.operation=equals&1_group.propertyvalues.property=.%2fjcr%3acontent%2fmetadata%2fdc%3aformat&2_group.relativedaterange.lowerBound=-1M&2_group.relativedaterange.property=jcr%3acontent%2fjcr%3alastModified&3_group.daterange.lowerBound=2018-8-1&3_group.daterange.property=jcr%3acreated&3_group.daterange.upperBound=2018-8-31&5_group.0_path=%2fcontent%2fdam%2fasset-share-commons%2fen%2fpublic&5_group.1_path=%2fcontent%2fdam%2fwe-retail%2fen&5_group.p.or=true&6_group.mainasset=true&7_group.property=jcr%3acontent%2fcontentFragment&7_group.property.operation=not&9_group.orderby=%40jcr%3acontent%2fjcr%3alastModified&p.guessTotal=100&p.limit=24&p.offset=0&type=dam%3aAsset
25.08.2018 19:52:39.211 *DEBUG* [0:0:0:0:0:0:0:1 [1535241159195] GET /content/asset-share-commons/en/light.results.html HTTP/1.1] com.day.cq.search.impl.builder.QueryImpl executing query (predicate tree):
ROOT=group: limit=24, guessTotal=100, offset=0[
    {type=type: type=dam:Asset}
    {1_group=group: [
        {propertyvalues=propertyvalues: property=./jcr:content/metadata/dc:format, operation=equals, 0_values=image/jpeg,image/gif,image/png,image/tiff,image/vnd.adobe.photoshop,application/postscript}
    ]}
    {2_group=group: [
        {relativedaterange=relativedaterange: property=jcr:content/jcr:lastModified, lowerBound=-1M}
    ]}
    {3_group=group: [
        {daterange=daterange: property=jcr:created, lowerBound=2018-8-1, upperBound=2018-8-31}
    ]}
    {5_group=group: or=true[
        {0_path=path: path=/content/dam/asset-share-commons/en/public}
        {1_path=path: path=/content/dam/we-retail/en}
    ]}
    {6_group=group: [
        {mainasset=mainasset: mainasset=true}
    ]}
    {7_group=group: [
        {property=property: property=jcr:content/contentFragment, operation=not}
    ]}
    {9_group=group: [
        {orderby=orderby: orderby=@jcr:content/jcr:lastModified}
    ]}
    {10_group=group: [
        {orderby=orderby: sort=desc}
    ]}
]
```
Note that the created relative date range works for me as well here.

![image](https://user-images.githubusercontent.com/1451868/44623594-bb92e480-a8a0-11e8-8cef-e0339c435ffe.png)

**Comment by Shaihuludus** [comment]
No, I tried only on 6.4.1 as this version is in my project requirements. The key to reproduction of this problem is to use only 3rd filter. In your example it would be CREATED, other filters shouldn't be used. 
It is possible that 6.4.1 has some issues with predicates. I will try 6.4.0 tommorow morning

**Comment by godanny86** [comment]
+1 for this bug. I'm able to reproduce on 6.4 GA.

When the only filter selected is the 3rd position it seems that the Search Paths (from the Search results component) on the server side is injected into the 3rd group. Since p.or=true for that predicate group the filter essentially never gets applied. 


```
6.08.2018 13:55:44.299 *DEBUG* [0:0:0:0:0:0:0:1 [1535306144281] GET /content/asset-share-commons/en/light.results.html HTTP/1.1] com.adobe.aem.commons.assetshare.search.providers.impl.QuerySearchProviderImpl Query Builder Parameters: 
3_group.0_path = /content/dam/asset-share-commons/en/public
3_group.1_path = /content/dam/we-retail/en
3_group.p.or = true
3_group.propertyvalues.0_values = properties:style/monochrome/grayscale
3_group.propertyvalues.operation = equals
3_group.propertyvalues.property = ./jcr:content/metadata/cq:tags
4_group.mainasset = true
5_group.property = jcr:content/contentFragment
5_group.property.operation = not
7_group.orderby = @jcr:content/jcr:lastModified
8_group.orderby.sort = desc
p.guessTotal = 100
p.limit = 24
p.offset = 0
type = dam:Asset
...

26.08.2018 13:55:44.300 *DEBUG* [0:0:0:0:0:0:0:1 [1535306144281] GET /content/asset-share-commons/en/light.results.html HTTP/1.1] com.day.cq.search.impl.builder.QueryImpl executing query (URL):
3_group.0_path=%2fcontent%2fdam%2fasset-share-commons%2fen%2fpublic&3_group.1_path=%2fcontent%2fdam%2fwe-retail%2fen&3_group.p.or=true&3_group.propertyvalues.0_values=properties%3astyle%2fmonochrome%2fgrayscale&3_group.propertyvalues.operation=equals&3_group.propertyvalues.property=.%2fjcr%3acontent%2fmetadata%2fcq%3atags&4_group.mainasset=true&5_group.property=jcr%3acontent%2fcontentFragment&5_group.property.operation=not&7_group.orderby=%40jcr%3acontent%2fjcr%3alastModified&8_group.orderby.sort=desc&p.guessTotal=100&p.limit=24&p.offset=0&type=dam%3aAsset
26.08.2018 13:55:44.300 *DEBUG* [0:0:0:0:0:0:0:1 [1535306144281] GET /content/asset-share-commons/en/light.results.html HTTP/1.1] com.day.cq.search.impl.builder.QueryImpl executing query (predicate tree):
ROOT=group: limit=24, guessTotal=100, offset=0[
    {type=type: type=dam:Asset}
    {3_group=group: or=true[
        {propertyvalues=propertyvalues: property=./jcr:content/metadata/cq:tags, operation=equals, 0_values=properties:style/monochrome/grayscale}
        {0_path=path: path=/content/dam/asset-share-commons/en/public}
        {1_path=path: path=/content/dam/we-retail/en}
    ]}
    {4_group=group: [
        {mainasset=mainasset: mainasset=true}
    ]}
    {5_group=group: [
        {property=property: property=jcr:content/contentFragment, operation=not}
    ]}
    {7_group=group: [
        {orderby=orderby: orderby=@jcr:content/jcr:lastModified}
    ]}
    {8_group=group: [
        {orderby=orderby: sort=desc}
    ]}
]
```

Seems like the

**Comment by godanny86** [comment]
@Shaihuludus do you mind verifying on the latest build of develop? PR #244 is intended to fix this.

**Comment by Shaihuludus** [comment]
It seems it works fine ;)

**Comment by davidjgonzalez** [comment]
released in v1.6.4

**Comment by pabloram23** [comment]
Hi @davidjgonzalez Is this issue re-appearing in newest versions of Asset Share Commons? I am using 3.2.0, upgraded to 3.2.4 and still facing the third filter problem that the `group.p.or = true` disrupts the facets and provide un-accurate results.

Please give advice on it. I can provide more info if required.
Thanks.
